src/main/kotlin/app/revanced/patches/tumblr/ads/DisableDashboardAds.kt

```diff
@@ -1,33 +1,40 @@
-import app.revanced.patcher.extensions.InstructionExtensions.replaceInstruction
 import app.revanced.patcher.patch.BytecodePatch
 import app.revanced.patcher.patch.annotation.CompatiblePackage
 import app.revanced.patcher.patch.annotation.Patch
-import app.revanced.patches.tumblr.ads.fingerprints.AdWaterfallFingerprint
-import com.android.tools.smali.dexlib2.iface.instruction.OneRegisterInstruction
+import app.revanced.patches.tumblr.timelinefilter.TimelineFilterPatch
 
 @Patch(
     name = "Disable dashboard ads",
     description = "Disables ads in the dashboard.",
-    compatiblePackages = [CompatiblePackage("com.tumblr")]
+    compatiblePackages = [CompatiblePackage("com.tumblr")],
+    dependencies = [TimelineFilterPatch::class]
 )
 @Suppress("unused")
-object DisableDashboardAds : BytecodePatch(
-    setOf(AdWaterfallFingerprint)
-) {
-    override fun execute(context: BytecodeContext) = AdWaterfallFingerprint.result?.let {
-        it.scanResult.stringsScanResult!!.matches.forEach { match ->
-            // We just replace all occurrences of "client_side_ad_waterfall" with anything else
-            // so the app fails to handle ads in the timeline elements array and just skips them.
-            // See AdWaterfallFingerprint for more info.
-            val stringRegister = it.mutableMethod.getInstruction<OneRegisterInstruction>(match.index).registerA
-            it.mutableMethod.replaceInstruction(
-                match.index, "const-string v$stringRegister, \"dummy\""
-            )
-        }
-    } ?: throw AdWaterfallFingerprint.exception
+object DisableDashboardAds : BytecodePatch() {
+    override fun execute(context: BytecodeContext)  {
+        // Called "client_side_ad_waterfall" in api response
+        TimelineFilterPatch.addObjectTypeFilter("CLIENT_SIDE_MEDIATION")
+        // Called "backfill_ad" in api response
+        TimelineFilterPatch.addObjectTypeFilter("GEMINI_AD")
+
+        // The below object types weren't actually spotted in the wild in testing, but they are valid Object types
+        // and their names clearly indicate that they are ads, so we just block them anyway,
+        // just in case they will be used in the future.
+
+        // Called "nimbus_ad" in api response
+        TimelineFilterPatch.addObjectTypeFilter("NIMBUS_AD")
+        // Called "client_side_ad" in api response
+        TimelineFilterPatch.addObjectTypeFilter("CLIENT_SIDE_AD")
+        // Called "display_io_interscroller" in api response
+        TimelineFilterPatch.addObjectTypeFilter("DISPLAY_IO_INTERSCROLLER_AD")
+        // Called "display_io_headline_video" in api response
+        TimelineFilterPatch.addObjectTypeFilter("DISPLAY_IO_HEADLINE_VIDEO_AD")
+        // Called "facebook_biddable_sdk_ad" in api response
+        TimelineFilterPatch.addObjectTypeFilter("FACEBOOK_BIDDAABLE")
+        // Called "google_native_ad" in api response
+        TimelineFilterPatch.addObjectTypeFilter("GOOGLE_NATIVE")
```

Comment:
Reduce duplicate code:

arrayOf("strings").forEach { TimelineFilterPatch.addObjectTypeFilter(it) }